doc: update Operating in power save modes user guide (#10237)
nrfconnect/sdk-nrf: doc/nrf/working_with_nrf/nrf70/developing/powersave.rst

**********************

Target Wake Time is a new feature in Wi-Fi 6.
It allows devices to wake up at the negotiated times to transmit and receive data.
The access point and devices reach a TWT agreement that defines when a station is active and ready to receive and transmit data.

> TWT mechanism allows each device to negotiate its wake-up period with the AP to transmit and receive data packets.

Reviewer: ```suggestion
The TWT mechanism allows each device to negotiate its wake-up period with the AP to transmit and receive data packets.
```

> Stations will receive and transmit at TWT Service Period only and remain in the leep state for the rest of the time.

Reviewer 2: leep state??

> Stations will receive and transmit during the TWT Service Period only and remain in the sleep state for the rest of the time.

Reviewer: ```suggestion
Stations will only receive and transmit during the TWT Service Period and remain in the sleep state for the rest of the time.
```

* **TWT Wake Duration**

  * The amount of time that the TWT requesting device needs to be active to complete the frame exchanges during the TWT wake interval.
  * Valid range for duration is 1 ms to 256 ms.
    Lower value results in more power saving at the cost of potential loss of data.
    It is recommended not to go lower than 8 ms.

>     The application can choose lower value than this at the cost of losing application data in the network.

Reviewer: ```suggestion
    The application can choose a lower value than 8 ms at the cost of losing application data in the network.
```

  * The application must choose the right duration based on the traffic pattern.
  * Applications must take appropriate action if uplink or downlink traffic is more than anticipated.
    It could terminate the ongoing TWT session and negotiate a fresh session based on new requirements.

  * Downlink traffic:

    a. Predictability is the key for choosing the correct wake duration.
    #. Wake duration must be sufficient for the AP to schedule all the incoming traffic to devices.

>        AP needs to contend the channel for scheduling frames and frames will be dropped if duration is aggressive in busy channels.

Reviewer: ```suggestion
       AP needs to contend the channel for scheduling frames and frames will be dropped if the duration is aggressive in busy channels.
```

    #. AP will drop the device data if it cannot finish all transmission in the wake duration.

>        AP may buffer traffic till the next interval if sleep duration is in the order of 100 milliseconds.

Reviewer 3: ```suggestion
       AP may buffer traffic until the next interval if sleep duration is in the order of 100 milliseconds.
```

>        AP may buffer traffic until the next interval if sleep duration is in the order of 100 milliseconds.

Reviewer: ```suggestion
       AP may buffer traffic until the next interval if sleep duration is in the order of 100 ms.
```

       It will not buffer the device data if sleep duration is in the order of minutes and data will be lost.
    #. Devices are allowed to sleep after the wake duration and there is no mechanism to extend the wake duration based on downlink traffic.
       Device must be active during the wake duration even if there is no downlink traffic.

  * Uplink traffic:

    a. Trigger Enabled - When operating in Trigger Enabled mode, the nRF70 Series device:

       * schedules uplink traffic as a response to trigger frames from the AP.
       * expects the AP to schedule trigger frames in the wake duration.

>        * discards all pending frames after wake time expiry before entering the sleep state.

Reviewer 3: Let's use uplink consistently to avoid confusion / help the reader.
```suggestion
       * discards all pending uplink frames after wake time expiry before entering the sleep state.
```

    #. Non-trigger Enabled - When operating in Non-trigger Enabled mode, the nRF70 Series device:

       * schedules uplink traffic using the legacy channel contention.
       * tries to schedule all uplink traffic in the wake duration and discard pending frames.

>        * discards all pending frames after wake time expiry before entering the sleep state.

Reviewer 3: ```suggestion
       * discards all pending uplink frames after wake time expiry before entering the sleep state.
```

* **TWT Wake Interval**

  * Interval between successive TWT wake periods.
  * Valid range for duration is one ms to a few days.

>   * Application must choose the right interval based on the traffic in receive direction.

Reviewer 3: @udaynordic Why specifically in the receive direction ? Surely it is equally applicable in both directions ? And should we say "... based on the **expected** traffic".

Reviewer 4: Agree Mike. Its applicable for both directions.

* **TWT Wake Duration**

  * The amount of time that the TWT requesting device needs to be active to complete the frame exchanges during the TWT wake interval.
  * Valid range for duration is 1 ms to 256 ms.

>     Lower value gives more power saving at the cost of potential loss of data.

Reviewer 2: "results in more power saving", perhaps?

  * Applications must take appropriate action if uplink or downlink traffic is more than anticipated.
    It could terminate the ongoing TWT session and negotiate a fresh session based on new requirements.

  * Downlink traffic:

    a. Predictability is the key for choosing the correct wake duration.
    #. Wake duration must be sufficient for the AP to schedule all incoming traffic to devices.
       AP needs to contend the channel for scheduling frames and frames will be dropped if duration is aggressive in busy channels.
    #. AP will drop device data if it cannot finish all transmission in wake duration.
       AP may buffer traffic till the next interval if sleep duration is in the order of 100 milliseconds.
       It will not buffer device data if sleep duration is in the order of minutes and data will be lost.
    #. Devices are allowed to sleep after wake duration and there is no mechanism to extend wake duration based on downlink traffic.
       Device must be active during wake duration even if there is no downlink traffic.

  * Uplink traffic:

    a. Trigger Enabled - When operating in Trigger Enabled mode, the nRF70 Series device:

>        * will schedule uplink traffic as response to trigger frames from the AP.

Reviewer 2: can we use the same tense in all these? E.g future or present

The wake-up and sleep period of the device is aligned to DTIM period advertised in the AP beacon.

> The access point is in control of DTIM period and can be configured while setting up the network.

Reviewer 4: ```suggestion
The access point is in control of DTIM period and can be configured while setting up the network. Stations connected to the AP cannot set or request a change to this value
```

* maintains the Wi-Fi connection by responding to the keep alive packet exchange at any point of time.

TWT-based power save allows devices to sleep for longer intervals than the DTIM power save.
It is suitable for devices that have predictable periodic uplink or downlink traffic and do not have low latency requirements.
DTIM-based power save is more efficient for sleep intervals that are in the range of milliseconds to a few seconds.

> TWT will perform better if the sleep interval is in 10s of seconds.

Reviewer 3: ```suggestion
TWT will perform better if the sleep interval is in the 10s of seconds and above range.
```
